### PR TITLE
refactor and create relational test server

### PR DESCRIPTION
**Pull request by vikask-gs**
#### What type of PR is this?

Refactoring

#### What does this PR do / why is it needed ?

It refactors integration tests for various relational dbs, so that we can eventually have one module per db. Also, it create a lean relationalStore-test-server dependency, which can be used instead of bulky server dependency in integration tests.

#### Which issue(s) this PR fixes:

None.

#### Does this PR introduce a user-facing change?

No. Also, it is backward compatible.

### CODE REVIEW

**Comment by epsstan** [comment]
Please update the PR description using the PR template (Under .github).

File: legend-engine-xt-relationalStore-sqlserver-execution-tests/src/main/java/org/finos/legend/engine/authentication/SqlServerTestDatabaseAuthenticationFlowProvider.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import org.eclipse.collections.api.list.ImmutableList;
+import org.eclipse.collections.impl.factory.Lists;
+import org.finos.legend.engine.authentication.flows.SqlServerStaticWithUserPasswordFlow;
+import org.finos.legend.engine.authentication.provider.AbstractDatabaseAuthenticationFlowProvider;
+import org.finos.legend.engine.authentication.provider.DatabaseAuthenticationFlowProviderConfiguration;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.authentication.AuthenticationStrategy;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.specification.DatasourceSpecification;
+
+public class SqlServerTestDatabaseAuthenticationFlowProvider extends AbstractDatabaseAuthenticationFlowProvider
```
Comment (epsstan) [inline legend-engine-xt-relationalStore-sqlserver-execution-tests/src/main/java/org/finos/legend/engine/authentication/SqlServerTestDatabaseAuthenticationFlowProvider.java:25]: I don't remember the details now .. but I think we have a bug in the flow provider/selector singleton implementation.  Something like if you have multiple flow providers in the CP, depending on how you initialize them, LegendDefaultDatabaseAuthenticationFlowProvider.configure throws an exception.

Have you run the full test suite with this change ?

File: legend-engine-xt-relationalStore-test-server/pom.xml [hunk 2]
```diff
@@ -0,0 +1,255 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (epsstan) [inline legend-engine-xt-relationalStore-test-server/pom.xml:1]: I see what you're doing here .. moving db specific test code from the engine server module to a test module specific to db tests. 

Is this pattern only for relational connections ? For e.g will there be a serviceStore-test-server ?

File: legend-engine-xt-mastery-pure/pom.xml [hunk 3]
```diff
@@ -127,11 +127,6 @@
             <groupId>org.finos.legend.engine</groupId>
```
Comment (epsstan) [inline legend-engine-xt-mastery-pure/pom.xml:127]: This is not related to the relational refactoring .. right ? If yes, please move to a separate PR. This PR is big as is.

**Review by epsstan — CHANGES_REQUESTED** [review]
I took a first pass. Can you please write down in the PR description (using the PR template) what are the types of changes being introduced ?

File: .github/workflows/database-bigquery-integration-test.yml [hunk 4]
```diff
@@ -69,4 +69,4 @@ jobs:
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_INTEGRATION_USER1_SECRET_ACCESS_KEY }}
         run: |
           mvn -P run-bigquery-integration -pl legend-engine-xt-relationalStore-executionPlan-connection-tests -Dtest="ExternalIntegration*BigQuery" test
-          mvn -P run-bigquery-integration -pl legend-engine-xt-relationalStore-executionPlan-connection-authentication -Dtest="ExternalIntegration*BigQuery*" test
+          mvn -P run-bigquery-integration -pl legend-engine-xt-relationalStore-executionPlan-connection-authentication-default -Dtest="ExternalIntegration*BigQuery*" test
```
[the remark this answers is not in the document]
Comment (vikask-gs) [reply .github/workflows/database-bigquery-integration-test.yml:72]: Yes,  once we create module for bigquery tests, this would change again

File: .github/workflows/database-databricks-sql-generation-integration-test.yml [hunk 5]
```diff
@@ -58,4 +58,4 @@ jobs:
       - name: Run Sql Generation Tests
         env:
           MAVEN_OPTS: -Xmx4g
-        run: mvn -pl legend-engine-server -Dtest="Test_Relational_DbSpecific_Databricks_UsingPureClientTestSuite" test
+        run: mvn -pl legend-engine-xt-relationalStore-test-server -Dtest="Test_Relational_DbSpecific_Databricks_UsingPureClientTestSuite" test
```
[the remark this answers is not in the document]
Comment (vikask-gs) [reply .github/workflows/database-databricks-sql-generation-integration-test.yml:61]: Correct

File: legend-engine-test-server-shared/pom.xml [hunk 6]
```diff
@@ -15,51 +15,72 @@
  limitations under the License.
 -->
 
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
[the remark this answers is not in the document]
Comment (vikask-gs) [reply legend-engine-test-server-shared/pom.xml:18]: I created a new pom.xml for this module, but since I was also deleting a pom.xml for other module (conn-test-api), git treats it as renaming + change :) . I based this pom.xml on server module's pom.xml. If this is incorrect, then we need to change all pom.xml like server's which have this issue.

File: legend-engine-test-server-shared/pom.xml [hunk 7]
```diff
@@ -15,51 +15,72 @@
...
     <parent>
         <groupId>org.finos.legend.engine</groupId>
         <artifactId>legend-engine</artifactId>
         <version>3.14.5-SNAPSHOT</version>
     </parent>
     <modelVersion>4.0.0</modelVersion>
+    <artifactId>legend-engine-test-server-shared</artifactId>
```
[the remark this answers is not in the document]
Comment (vikask-gs) [reply legend-engine-test-server-shared/pom.xml:25]: Git's diff magic :) I created a new module and deleted one module. Git linked them together in diff

File: legend-engine-xt-relationalStore-sqlserver-execution-tests/pom.xml [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    <packaging>jar</packaging>
+
+    <build>
+        <plugins>
+            <!-- Block running the _server tests (They should only execute in Github/TeamCity as Integration Tests). Run only H2 tests-->
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <useSystemClassLoader>false</useSystemClassLoader>
+                    <excludes>
```
[the remark this answers is not in the document]
Comment (vikask-gs) [reply legend-engine-xt-relationalStore-sqlserver-execution-tests/pom.xml:37]: ok

File: legend-engine-xt-relationalStore-sqlserver-execution-tests/src/main/java/org/finos/legend/engine/authentication/SqlServerTestDatabaseAuthenticationFlowProvider.java (hunk 1 above)
Comment (vikask-gs) [reply legend-engine-xt-relationalStore-sqlserver-execution-tests/src/main/java/org/finos/legend/engine/authentication/SqlServerTestDatabaseAuthenticationFlowProvider.java:25]: SqlServerTest flow is only used in this module, and it does not conflicts with LegendDefault flow in any module. What sort of tests you have in mind? (Tc was green)

File: legend-engine-xt-relationalStore-test-server/pom.xml (hunk 2 above)
Comment (vikask-gs) [reply legend-engine-xt-relationalStore-test-server/pom.xml:1]: Yes, I see us creating serviceStore-test-server and so on in future (if it helps in integration testing for those stores)

File: legend-engine-xt-mastery-pure/pom.xml (hunk 3 above)
Comment (vikask-gs) [reply legend-engine-xt-mastery-pure/pom.xml:127]: Duplicate dependency declaration in this pom.xml. It was giving a warning during build, so removed it. Still want me remove it?
